app/src/server/utils/startTestJobs.ts

```diff
@@ -37,6 +68,6 @@ export const startTestJobs = async (fineTune: FineTune) => {
     skipDuplicates: true,
   });
   for (const entry of datasetEntries) {
-    await evaluateTestSetEntry.enqueue({ fineTuneId: fineTune.id, datasetEntryId: entry.id });
```

Comment:
We need to refactor the way this function works so it first figures out which outputs have actually been generated and then only does the `createMany`/`enqueue` on the ones that are actually missing. That'll make this way faster and less resource intensive.

It'll also provide a far better UX. I added a gpt-3.5 comparison 10 minutes ago for a dataset with 2500 test entries and 2 fine tuned models and it still hasn't started showing any of the 3.5 completions since it's still working its way through the 5000 no-op jobs for the existing fine-tunes.

Reply:
That's what I'm trying to do on line 35. I suspect a lot of the latency you saw was due to our previous step of calculating input tokens.